### ISSUE TITLE
AVISource: couldn't locate a decompressor for fourcc HFYU

### ISSUE THREAD

**Issue by nj2359** [staxrip/staxrip#1167]
StaxRip 2.3.0 - can't use AVISource
StaxRip works on my older system with a 4690k CPU
On my newer system, with an 8700k cpu, is where I get the error.
Both systems are running Win10 and have the prerequisites of .net and visual C++
The issue exists for both FFV1 and HuffYUV lossless encoded files.  I have not tested other formats.

Note*  FFVideoSource and DirectShowSource work and will load the video.
I do not want to use FFVideoSource, which is selected by default, because audio goes out of sync when deinterlacing with QTGMC.  DirectShowSource works and produces a good copy of my final encoded file to FFV1 .MOV with the audio still in sync.
However, I want to figure out why I can't use AVISource which will also keep the audio in sync, as tested on the working system.

The 8700k system had AviSynth+ 3.7.3 installed.  I uninstalled it in case it was somehow interfering with the embedded AviSynth in StaxRip, but that did not resolve the issue.
I've recopied the original StaxRip portable folder multiple times to ensure I'm testing with an unaltered copy.

In StaxRip, AviSynth Mode: is set to "Use Portable directly"
![image](https://github.com/staxrip/staxrip/assets/25471276/aef738e8-20bf-4d56-913f-4c9614cbd685)

Is StaxRip fully portable and self contained?  I'm not sure where else to go.  See log output below.
`
------------------------- System Environment -------------------------

StaxRip    : v2.30.0
Windows    : Windows 10 Enterprise 22H2 2009 (19041.1.amd64fre.vb_release.191206-1406)
Language   : English (United States)
CPU        : Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz
RAM        : 64GB
GPU        : NVIDIA GeForce RTX 3080
Resolution : 1920 x 1080
DPI        : 96
Code Page  : 1252

----------------------- Media Info Source File -----------------------

D:\TestVideo.avi

General
Complete name                : D:\TestVideo.avi
Format                       : AVI
Format/Info                  : Audio Video Interleave
Format profile               : OpenDML
Format settings              : WaveFormatEx
File size                    : 5.34 GiB
Duration                     : 14 min 9 s
Overall bit rate             : 54.0 Mb/s
Frame rate                   : 29.970 FPS

Video
ID                           : 0
Format                       : HuffYUV
Format version               : Version 2
Codec ID                     : HFYU
Duration                     : 14 min 9 s
Bit rate                     : 52.5 Mb/s
Width                        : 720 pixels
Height                       : 576 pixels
Display aspect ratio         : 5:4
Frame rate                   : 29.970 (30000/1001) FPS
Color space                  : YUV
Chroma subsampling           : 4:2:2
Bit depth                    : 8 bits
Scan type                    : Interlaced
Bits/(Pixel*Frame)           : 4.220
Stream size                  : 5.19 GiB (97%)

Audio
ID                           : 1
Format                       : PCM
Format settings              : Little / Signed
Codec ID                     : 1
Duration                     : 14 min 9 s
Bit rate mode                : Constant
Bit rate                     : 1 536 kb/s
Channel(s)                   : 2 channels
Sampling rate                : 48.0 kHz
Bit depth                    : 16 bits
Stream size                  : 156 MiB (3%)
Alignment                    : Aligned on interleaves
Interleave, duration         : 10  ms (0.29 video frame)
Interleave, preload duration : 49  ms

------------------------ Error opening source ------------------------

AVISource: couldn't locate a decompressor for fourcc HFYU
(D:\TestVideo_temp\TestVideo_source.avs, line 3)

AddAutoloadDir("C:\tools2\StaxRip-v2.30.0-x64_test\StaxRip-v2.30.0-x64\Apps\FrameServer\AviSynth\plugins")

AviSource("D:\TestVideo.avi", audio=false)`

**Comment by Dendraspis** [comment]
Which version of StaxRip are you using on the "working" system?

> Is StaxRip fully portable and self contained? I'm not sure where else to go.

Yes, no installation of `AviSynth+` required.

**Comment by nj2359** [comment]
Yes, I made sure to replicate the same environment as much as possible.  Same version 2.3.0, exact same source video file.  If there are other logs I should be pulling, let me know.  Enabling debugging does not produce any more in this particular log output, as it cannot get past loading the file.
I'd like to know where it is looking for the source filter and failing.

**Tested on a 3rd system and I have the same exact issue not being able to use AviSource 
AVISource: couldn't locate a decompressor for fourcc HFYU.**
**Things to note:**
klite codec pack is installed on all systems
the 2 problematic systems have adobe creative suite, premier pro, photoshop, lightroom classic
I do not want to uninstall adobe suite
I want to know how to pinpoint the cause without just trial and error
I'm not sure if or how other software will interfere with StaxRip, as it is portable.
I will try on a clean 4th machine eventually.

**AviSource filter is the only one keeping the audio in sync during deinterlacing, so I really want to get it working.**
I discovered **DirectShowSource does NOT** keep audio sync on longer videos when deinterlacing with QTGMC, when
converting from HuffYUV to FFV1, it becomes evident in longer videos as audio is out of sync toward end.
In my previous workflow with Virtualdub2, I would capture old VHS tapes and encode in HuffYUV lossles, but then I discovered the better FFV1 lossless, so I started using that.
So I've got a mix of HuffYUV and FFV1 source files
With Virtualdub2, I would then need to re-encode to ProRes422 without doing anything like deinterlacing, otherwise I'd lose audio sync.
I would then need to deinterlace the ProRes422 file and also encode the final output to ProRes422 as well.
This was the only way I could find to keep the audio in sync.
I want to eliminate the extra middle step of re-encoding and ideally keep the FFV1 format.
**So far, StaxRip using the AviSource Filter seems to be working the way I need, however, I need it to function on my faster machine.**

Comparing Virtualdub2 vs StaxRip for encoding jobs, StaxRip is nicer to work with by not having to manually edit the scripts for each file, so it's quicker and easier to queue up jobs
It is also better for anyone who is not very familiar with the scripting language, like myself, and being able to see how StaxRip builds the scripts is very helpful.
